stdlib/result.mc

```diff
@@ -348,6 +348,40 @@ utest
   ()
 with ()
 
+let _foldl : all w. all e. all a. all b.
```

Comment:
`_foldlM` to be consistent with the other functions, and include in the interface at the bottom.

Do consider if you can structure things in a different way if possible though, since this will very effectively suppress the ability to collect multiple errors, since we can't run a computation for a `b` if we already found a failure.

Reply:
I don't think it makes sense to gather multiple errors in most contexts. If the new value of the accumulator can not be computed for some element in the list, then I would argue the computation is undefined for all further computations since there is no accumulator. 

E.g., if you are traversing the spine of a program like this:
```
let f = ...
let x = ...
let y = some error here
let z = f x y
```
If we have an error at the computation of `y`, then the accumulator will maybe not know enough about `y` and we will get another error at `z` which can then also propegate upwards. So you get a ton of irrelevant errors instead of the single **actual** error.

Reply:
I meant more to consider if it is possible to formulate things such that they're more independent of each other, so you can report more errors. For example, in `tool.mc` there's an environment being passed around with type `Map Name (Either (Res TypeInfo) (Res TokenInfo))` (where `Res` is an alias around `Result`), so that even if a definition is faulty, I can still keep track of whether it's a type or a token and error if it's used incorrectly. Key point being that the type isn't `Res (Map Name (Either TypeInfo TokenInfo))`, since that's less precise and allows fewer errors. It's rarely *truly* the case that a particular value depends on the entirety of an accumulator, it's just sometimes inconvenient to insert `Result` at useful places.